packages/esm-reports-app/src/components/reports.resource.tsx

```diff
@@ -0,0 +1,229 @@
+import { openmrsFetch, Session, FetchResponse, openmrsObservableFetch } from '@openmrs/esm-framework';
+import { Observable } from 'rxjs';
+import useSWR from 'swr';
+import moment from 'moment';
+import { ReportDefinition } from '../types/report-definition';
+import { ReportDesign } from '../types/report-design';
+import { ReportRequest } from '../types/report-request';
+
+interface ReportModel {
+  reportName: string;
+  status: string;
+  requestedBy: string;
+  requestedByUserUuid: string;
+  requestedOn: string;
+  outputFormat: string;
+  parameters: any;
+  id: string;
+  evaluateCompleteDatetime: string;
+  schedule: string;
+}
+
+interface ScheduledReportModel {
+  reportDefinitionUuid: string;
+  reportRequestUuid: string;
+  name: string;
+  schedule: string;
+}
+
+export function useLocations() {
+  const apiUrl = `/ws/rest/v1/location?tag=Login+Location`;
+
+  const { data } = useSWR<{ data: { results: Array<any> } }, Error>(apiUrl, openmrsFetch);
+
+  return {
+    locations: data ? data?.data?.results : [],
+  };
+}
+
+export function useReports(statuses: string, pageNumber: number, pageSize: number, sortBy?: string): any {
+  const reportsUrl =
+    `/ws/rest/v1/reportingrest/reportRequest?status=${statuses}&startIndex=${pageNumber}&limit=${pageSize}&totalCount=true` +
+    (sortBy ? `&sortBy=${sortBy}` : '');
+
+  const { data, error, isValidating, mutate } = useSWR<{ data: { results: Array<any>; totalCount: number } }, Error>(
+    reportsUrl,
+    openmrsFetch,
+  );
+
+  const reports = data?.data?.results;
+  const totalCount = data?.data?.totalCount;
+  const reportsArray: Array<any> = reports ? [].concat(...reports.map((report) => mapReportResults(report))) : [];
+
+  return {
+    reports: reportsArray,
+    reportsTotalCount: totalCount,
+    isError: error,
```

Comment:
```suggestion
   error,
```

This should just be `error`. `isError` incorrectly implies that the value is a boolean. It's not - it is `Error | undefined`.

Reply:
right, fixed